RevenueCatUI/CustomerCenter/Views/PromotionalOfferView.swift

```diff
@@ -131,24 +136,35 @@ struct PromoOfferButtonView: View {
             let mainTitle = discount.localizedPricePerPeriodByPaymentMode(.current)
             let localizedProductPricePerPeriod = product.localizedPricePerPeriod(.current)
 
-            Button(action: {
-                Task {
-                    await viewModel.purchasePromo()
+            AsyncButton {
+                withAnimation(.easeInOut(duration: 0.3)) {
+                    loadingState = true
+                }
+                await viewModel.purchasePromo()
```

Comment:
Just to make sure but this shouldn't trigger any errors right?

Reply:
we are catching the error, and not doing anything with it atm, but that's from before this change, so that improvement should come in a different PR